### PR TITLE
Move proxies migration

### PR DESCRIPTION
**Pull request by apopiak**
This PR moves the time delayed proxy migration from `on_runtime_upgrade` to a freestanding function in order to not trigger the migration on every runtime upgrade that uses this pallet.

### CODE REVIEW

File: frame/proxy/src/lib.rs [hunk 1]
```diff
@@ -675,3 +659,23 @@ impl<T: Trait> Module<T> {
...
+						proxy_type: t,
+						delay: Zero::zero(),
+					})
+					.collect::<Vec<_>>(),
+				deposit,
+			))
+		);
+		T::MaximumBlockWeight::get()
+	}
+}
```
Comment (shawntabrizi) [inline frame/proxy/src/lib.rs:681]:
```suggestion
}

```

**Review by shawntabrizi — APPROVED** [review]
this looks right to me. @bkchr @gnunicorn ?

File: frame/proxy/src/lib.rs [hunk 2]
```diff
@@ -675,3 +659,23 @@ impl<T: Trait> Module<T> {
 		Self::deposit_event(RawEvent::ProxyExecuted(e.map(|_| ()).map_err(|e| e.error)));
 	}
 }
+
+pub mod migration {
```
Comment (bkchr) [inline frame/proxy/src/lib.rs:663]: Some docs would be nice

File: frame/proxy/src/lib.rs [hunk 3]
```diff
@@ -675,3 +659,23 @@ impl<T: Trait> Module<T> {
...
+	use super::*;
+
+	pub fn migrate_to_time_delayed_proxies<T: Trait>() -> Weight {
```
Comment (bkchr) [inline frame/proxy/src/lib.rs:666]: Here as well

**Review by bkchr — APPROVED** [review]

**Review by gavofyork — APPROVED** [review]

File: frame/proxy/src/lib.rs (hunk 2 above)
Comment (shawntabrizi) [reply frame/proxy/src/lib.rs:664]: done

File: frame/proxy/src/lib.rs (hunk 3 above)
Comment (shawntabrizi) [reply frame/proxy/src/lib.rs:673]: done

**Comment by apopiak** [comment]
thanks for the docs Shawn!

**Comment by apopiak** [comment]
bot merge

**Comment by apopiak** [comment]
**Note:** If using the introduced migration function, you should use storage guards to make sure it
only runs once.

**Pull request merged**